elements/pfe-switch/src/pfe-switch.js

```diff
@@ -0,0 +1,178 @@
+import PFElement from "../../pfelement/dist/pfelement.js";
+
+class PfeSwitch extends PFElement {
+  static get tag() {
+    return "pfe-switch";
+  }
+
+  get schemaUrl() {
+    return "pfe-switch.json";
+  }
+
+  get templateUrl() {
+    return "pfe-switch.html";
+  }
+
+  get styleUrl() {
+    return "pfe-switch.scss";
+  }
+
+  static get events() {
+    return {
+      change: `${this.tag}:change`
+    };
+  }
+
+  get pfeChecked() {
+    return this.getAttribute("pfe-checked");
+  }
+
+  // Declare the type of this component
+  static get PfeType() {
+    return PFElement.PfeTypes.Content;
+  }
+
+  constructor() {
+    super(PfeSwitch, { type: PfeSwitch.PfeType });
+    this._updateSwitchState = this._updateSwitchState.bind(this);
+    this.settings = {};
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const $pfeSwitch = this.shadowRoot.querySelector(".pfe-switch");
+
+    // Get settings from attributes
+    const settingAttributes = [
```

Comment:
This is done for you by importing the schema.  Any attributes defined in your schema will be attached to your component for you by logic built into `pfelement`.  You can access that information using `this.props["message-on"]` which will list out the object for that attribute.  Each attribute has title, type, enum (if defined), default (if defined), a boolean prefixed value (does it have `pfe` at the beginning of it), and the actual value assigned (if the user has added that attribute to the component).